packages/material/src/util/debounce.ts

```diff
@@ -0,0 +1,16 @@
+import { debounce } from 'lodash';
+import { useState, useCallback } from 'react'
+
+
+const eventToValue = (ev: any) => ev.target.value;
+export const useDebouncedChange = (handleChange: (path: string, value: any) => void, defaultValue: any, data: any, path: string, eventToValueFunction: (ev: any) => any = eventToValue, timeout = 300): [any, React.ChangeEventHandler, () => void] => {
+  const [input, setInput] = useState(data || defaultValue);
+  const debouncedUpdate = useCallback(debounce((newValue: string) => handleChange(path, newValue), timeout), [handleChange]);
+  const onChange = useCallback((ev: any) => {
+    const newValue = eventToValueFunction(ev);
+    setInput(newValue);
+    debouncedUpdate(newValue);
+  }, [debouncedUpdate]);
+  const onClear = useCallback(() => { setInput(defaultValue); handleChange(path, undefined) }, [handleChange]);
```

Comment:
Now that we're talking about "exhaustive-deps" in the other comment chain, I just realized that `defaultValue` and `path` is missing here in the dependencies.